retrofit-converters/moshi/src/main/java/retrofit/MoshiRequestBodyConverter.java

```diff
@@ -18,39 +18,25 @@
 import com.squareup.moshi.JsonAdapter;
 import com.squareup.okhttp.MediaType;
 import com.squareup.okhttp.RequestBody;
-import com.squareup.okhttp.ResponseBody;
 import java.io.IOException;
 import okio.Buffer;
-import okio.BufferedSource;
 
-final class MoshiConverter<T> implements Converter<T> {
+final class MoshiRequestBodyConverter<T> implements Converter<T, RequestBody> {
   private static final MediaType MEDIA_TYPE = MediaType.parse("application/json; charset=UTF-8");
 
-  private JsonAdapter<T> adapter;
+  private final JsonAdapter<T> adapter;
 
-  MoshiConverter(JsonAdapter<T> adapter) {
+  MoshiRequestBodyConverter(JsonAdapter<T> adapter) {
     this.adapter = adapter;
   }
 
-  @Override public T fromBody(ResponseBody body) throws IOException {
-    BufferedSource source = body.source();
-    try {
-      return adapter.fromJson(source);
-    } finally {
-      try {
-        source.close();
-      } catch (IOException ignored) {
-      }
-    }
-  }
-
-  @Override public RequestBody toBody(T value) {
+  @Override public RequestBody convert(T value) throws IOException {
     Buffer buffer = new Buffer();
```

Comment:
Also non-streaming. If I recall correctly this is because `value` might be mutable, right?